lib/auth/v2/headerAuthCheck.ts

```diff
@@ -16,14 +18,13 @@ export function check(
     const headers = request.headers;
 
     const token = headers['x-amz-security-token'];
-    if (token && !constants.iamSecurityToken.pattern.test(token)) {
+    if (token && !iamSecurityTokenPattern.test(token)) {
         log.debug('invalid security token', { token });
         return { err: errors.InvalidToken };
     }
 
     // Check to make sure timestamp is within 15 minutes of current time
-    let timestamp = headers['x-amz-date'] ?
-        headers['x-amz-date'] : headers.date;
+    let timestamp = headers['x-amz-date'] || headers.date;
     timestamp = Date.parse(timestamp);
```

Comment:
Make the variable `const` and avoid changing type:

```suggestion
    const timestamp = Date.parse(headers['x-amz-date'] || headers.date);
```